src/audio/kpb.c

```diff
@@ -721,8 +721,11 @@ static int kpb_buffer_data(struct comp_dev *dev,
 
 	tracev_kpb_with_ids(dev, "kpb_buffer_data()");
 
-	if (kpb->state != KPB_STATE_RUN && kpb->state != KPB_STATE_DRAINING)
+	if (kpb->state != KPB_STATE_RUN && kpb->state != KPB_STATE_DRAINING) {
```

Comment:
@mrajwa ok, so we now have a finite state machine, so we need to have a state transition diagram like we do for general purpose components. Can you add this to make this easier to track and understand.
Another useful thing would be (after the diagram is added) is to add some conditional DEBUG code in ```kpb_change_state()``` that validates each state transition i.e. will emit trace_error if invalid transition is attempted.

Reply:
Totally, where would you like me to add this state transition diagram?

Reply:
yes please